Sources/Data Model/Audience/SemanticVersion.swift

```diff
@@ -0,0 +1,61 @@
+/****************************************************************************
+* Copyright 2020, Optimizely, Inc. and contributors                        *
+*                                                                          *
+* Licensed under the Apache License, Version 2.0 (the "License");          *
+* you may not use this file except in compliance with the License.         *
+* You may obtain a copy of the License at                                  *
+*                                                                          *
+*    http://www.apache.org/licenses/LICENSE-2.0                            *
+*                                                                          *
+* Unless required by applicable law or agreed to in writing, software      *
+* distributed under the License is distributed on an "AS IS" BASIS,        *
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+* See the License for the specific language governing permissions and      *
+* limitations under the License.                                           *
+***************************************************************************/
+
+import Foundation
+
+typealias SemanticVersion = String
+/*
+ This comparison is ported from the Optimizely web version of semantic version compare.
+ Full testing in SemanticVersionTests.
+ */
+extension SemanticVersion {
+    func compareVersion(targetedVersion: SemanticVersion?) -> Int {
+        guard let targetedVersion = targetedVersion else {
+            // Any version.
+            return 0
+          }
+
+          // Expect a version string of the form x.y.z
+        let targetedVersionParts = targetedVersion.split(separator: ".");
+        let versionParts = self.split(separator: ".");
+
+        // Up to the precision of targetedVersion, expect version to match exactly.
+        for (idx, _) in targetedVersionParts.enumerated() {
+            if versionParts.count <= idx {
+              return -1;
+            } else if !String(versionParts[idx]).isNumber {
+                //Compare strings
+                if versionParts[idx] != targetedVersionParts[idx] {
+                    return -1;
```

Comment:
Is this correct?  I expect: 3.1.2-beta > 3.1.0, but this returns opposite.
When it's required for a client to turn on a feature for version > 3.1.0, this rule should be true to all beta version after the target as well.

Reply:
yeah I need to know the same thing. 
What about comparing `alpha` and `beta`